site/src/Map.jsx

```diff
@@ -1,85 +1,93 @@
+const MAP_STYLE =
+  "https://basemaps.cartocdn.com/gl/positron-gl-style/style.json";
+const DARK_MAP_STYLE =
+  "https://basemaps.cartocdn.com/gl/dark-matter-gl-style/style.json";
 
-const PLACES_PMTILES_URL = 'pmtiles://https://data.source.coop/protomaps/overture/2024-04-16-beta.0/places.pmtiles';
+const PLACES_PMTILES_URL =
+  "pmtiles://https://data.source.coop/protomaps/overture/2024-04-16-beta.0/places.pmtiles";
 const PLACES_MAP_STYLE = {
-  'id': 'places',
-  'type': 'circle',
-  'source': 'places',
-  'source-layer': 'places',
-  'filter': [">=", ["get", "confidence"], 0.0],
-  'paint': {
-    'circle-color':
-      ['case',
-        ['==', ['get', '@category'], 'beauty_salon'], '#fb9a99',
-        ['==', ['get', '@category'], 'hotel'], '#33a02c',
-        ['==', ['get', '@category'], 'landmark_and_historical_building'], '#a6cee3',
-        ['==', ['get', '@category'], 'professional_services'], '#fdbf6f',
-        ['==', ['get', '@category'], 'shopping'], '#e31a1c',
-        ['==', ['get', '@category'], 'restaurant'], '#1f78b4',
-        ['==', ['get', '@category'], 'school'], '#ff7f00',
-        ['==', ['get', '@category'], 'accommodation'], '#b2df8a',
-        '#cab2d6'
-      ],
-    'circle-radius': [
-      'interpolate',
-      ['exponential', 2],
-      ['zoom'],
-      0, 1,
-      19, 8
+  id: "places",
+  type: "circle",
+  source: "places",
+  "source-layer": "places",
+  filter: [">=", ["get", "confidence"], 0.0],
+  paint: {
+    "circle-color": [
+      "case",
+      ["==", ["get", "@category"], "beauty_salon"],
+      "#fb9a99",
+      ["==", ["get", "@category"], "hotel"],
+      "#33a02c",
+      ["==", ["get", "@category"], "landmark_and_historical_building"],
+      "#a6cee3",
+      ["==", ["get", "@category"], "professional_services"],
+      "#fdbf6f",
+      ["==", ["get", "@category"], "shopping"],
+      "#e31a1c",
+      ["==", ["get", "@category"], "restaurant"],
+      "#1f78b4",
+      ["==", ["get", "@category"], "school"],
+      "#ff7f00",
+      ["==", ["get", "@category"], "accommodation"],
+      "#b2df8a",
+      "#cab2d6",
     ],
-    'circle-stroke-width': [
-      'interpolate',
-      ['exponential', 2],
-      ['zoom'],
-      12, 0,
-      14, 2
+    "circle-radius": ["interpolate", ["exponential", 2], ["zoom"], 0, 1, 19, 8],
+    "circle-stroke-width": [
+      "interpolate",
+      ["exponential", 2],
+      ["zoom"],
+      12,
+      0,
+      14,
+      2,
     ],
-    'circle-stroke-color': 'black'
-  }
+    "circle-stroke-color": "black",
+  },
 };
 
-
 const INITIAL_VIEW_STATE = {
-  latitude: 51.0500,
+  latitude: 51.05,
   longitude: 3.7303,
   zoom: 16,
   bearing: 0,
-  pitch: 0
+  pitch: 0,
 };
-export default function Map({mode}) {
-
-  const [pmTilesReady, setPmTilesReady] = useState(false)
-  const [cursor, setCursor] = useState('auto');
-  const [mapEntity, setMapEntity] = useState({});
+export default function Map({ mode, mapEntity, setMapEntity }) {
```

Comment:
Only contentful change: extracting the mapEntity state. All else is formatting changes